### PR TITLE
By clicking on the picture open new screen

### PR DESCRIPTION
**Pull request by IvanJanuary**

### CODE REVIEW

File: PersonDataRequest/PersonDataRequest/Controller/DataViewController.swift [hunk 1]
```diff
@@ -58,7 +61,7 @@ class DataViewController: UIViewController {
                 
             case .failure(let error):
                 DispatchQueue.main.async {
-                    self?.errorAlert(with: error)
+                    self?.errorAlert(with: error, completion: {self?.queryUserData()})
```
Comment (Lana-Resquit) [inline PersonDataRequest/PersonDataRequest/Controller/DataViewController.swift]: Please add [weak self] in completion

File: PersonDataRequest/PersonDataRequest/Controller/DataViewController.swift [hunk 2]
```diff
@@ -20,6 +20,9 @@ class DataViewController: UIViewController {
     @IBOutlet weak var activityLabel: UILabel!
     @IBOutlet weak var collectionView: UICollectionView!
     
+    
+    
+
```
Comment (Lana-Resquit) [inline PersonDataRequest/PersonDataRequest/Controller/DataViewController.swift]: Remove these extra lines

File: PersonDataRequest/PersonDataRequest/Controller/DataViewController.swift [hunk 3]
```diff
@@ -94,7 +99,7 @@ class DataViewController: UIViewController {
              }
          case .failure(let error):
              DispatchQueue.main.async {
-                 self.errorAlert(with: error)
+                 self.errorAlert(with: error) { self.queryUserData()}
```
Comment (Lana-Resquit) [inline PersonDataRequest/PersonDataRequest/Controller/DataViewController.swift]: Please add [weak self] in completion here as well

File: PersonDataRequest/PersonDataRequest/Controller/DetailViewController.swift [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+                    DispatchQueue.main.async {
+                        self?.errorAlert(with: error, completion: {})
+                    }
+                }
+            }
+        }
+    }
+}
+
+extension UIViewController {
```
Comment (Lana-Resquit) [inline PersonDataRequest/PersonDataRequest/Controller/DetailViewController.swift]: Please create a separate file for this extension

File: PersonDataRequest/PersonDataRequest/Controller/DataViewController.swift (hunk 2 above)
Comment (IvanJanuary) [reply PersonDataRequest/PersonDataRequest/Controller/DataViewController.swift]: Done

File: PersonDataRequest/PersonDataRequest/Controller/DataViewController.swift (hunk 3 above)
Comment (IvanJanuary) [reply PersonDataRequest/PersonDataRequest/Controller/DataViewController.swift]: Added

File: PersonDataRequest/PersonDataRequest/Controller/DataViewController.swift (hunk 1 above)
Comment (IvanJanuary) [reply PersonDataRequest/PersonDataRequest/Controller/DataViewController.swift]: Added

File: PersonDataRequest/PersonDataRequest/Controller/DetailViewController.swift (hunk 4 above)
Comment (IvanJanuary) [reply PersonDataRequest/PersonDataRequest/Controller/DetailViewController.swift]: Created

**Pull request merged**